src/rdkafka_admin.c

```diff
@@ -839,6 +957,114 @@ rd_kafka_admin_worker (rd_kafka_t *rk, rd_kafka_q_t *rkq, rd_kafka_op_t *rko) {
+                rd_kafka_enq_once_new(rko, RD_KAFKA_REPLYQ(rk->rk_ops, 0));
+
+        /* Set up replyq */
+        rd_kafka_set_replyq(&rko->rko_u.admin_fanout.replyq,
+                            rkqu->rkqu_q, 0);
+
+        return rko;
+}
+
+static rd_kafka_op_res_t
+rd_kafka_admin_fanout_worker (rd_kafka_t *rk, rd_kafka_q_t *rkq, rd_kafka_op_t *rko) {
+        const char *name = rd_kafka_op2str(rko->rko_type);
+        rd_kafka_event_t* evt;
+        rd_kafka_op_t* rko_result;
+        int i;
+
+        if (rd_kafka_terminating(rk)) {
+                rd_kafka_dbg(rk, ADMIN, name,
+                             "%s worker called: "
+                             "handle is terminating: %s",
+                             name,
+                             rd_kafka_err2str(rko->rko_err));
+                goto destroy;
+        }
+
+        if (rko->rko_err == RD_KAFKA_RESP_ERR__DESTROY)
+                goto destroy; /* rko being destroyed (silent) */
+
+        rd_kafka_dbg(rk, ADMIN, name,
+                     "%s worker called with %"PRIusz" requests outstanding: %s",
+                     name,
+                     rko->rko_u.admin_fanout.outstanding,
+                     rd_kafka_err2str(rko->rko_err));
+
+        rd_assert(thrd_is_current(rko->rko_rk->rk_thread));
+
+        while ((evt = rd_kafka_queue_poll(rko->rko_u.admin_fanout.responseq, 0))) {
+                rd_assert(rko->rko_u.admin_fanout.outstanding > 0);
+                --rko->rko_u.admin_fanout.outstanding;
+
+                rko->rko_u.admin_fanout.cbs->partial_response(rko, evt);
+
+                rd_kafka_event_destroy(evt);
+        }
+
+        if (rko->rko_u.admin_fanout.outstanding > 0) {
+                rd_kafka_dbg(rk, ADMIN, name,
+                             "%s worker still waiting for %"PRIusz" responses",
+                             name,
+                             rko->rko_u.admin_fanout.outstanding);
+                rd_kafka_enq_once_reenable(rko->rko_u.admin_fanout.eonce,
+                                           rko,
+                                           RD_KAFKA_REPLYQ(rk->rk_ops, 0));
+                return RD_KAFKA_OP_RES_KEEP; /* wait for outstanding requests to return */
+        }
+
+        rd_kafka_dbg(rk, ADMIN, name,
+                     "%s worker received all responses, enqueuing result",
+                     name);
+
+        rko_result = rd_kafka_admin_fanout_result_new(rko);
+
+        for (i = 0; i < rd_list_cnt(&rko->rko_u.admin_fanout.responses); i++) {
+                const void *elem = rd_list_elem(&rko->rko_u.admin_fanout.responses, i);
+                rd_list_add(&rko_result->rko_u.admin_result.results,
+                            rko->rko_u.admin_fanout.cbs->copy_result(elem));
+        }
+
+        /* Enqueue result on application queue, we're done. */
+        rd_kafka_replyq_enq(&rko->rko_u.admin_fanout.replyq, rko_result,
+                            rko->rko_u.admin_fanout.replyq.version);
+
```

Comment:
add `/*FALLTHRU*/` comment to make it clear that we do indeed want to fall thru to the destroy and we didnt just miss a return.